### PR TITLE
Add schema for plugin

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: packages/flutter_tools/schema/pubspec_yaml.json [hunk 1]
```diff
@@ -39,6 +39,13 @@
                             }
                         }
                     }
+                },
+                "plugin": {
+                    "type": "object",
+                    "properties": {
```
Comment (jakobr-google) [inline packages/flutter_tools/schema/pubspec_yaml.json]: Do we want to add `"additionalProperties": false,`?

File: packages/flutter_tools/schema/pubspec_yaml.json [hunk 2]
```diff
@@ -39,6 +39,13 @@
...
+                        "androidPackage": {"type": "string"},
```
Comment (jakobr-google) [inline packages/flutter_tools/schema/pubspec_yaml.json]: Maybe add spaces after `{` and before `}` to match the existing style.

**Review by jakobr-google — APPROVED** [review]
LGTM.

**Comment by szakarias** [comment]
Fixes https://github.com/flutter/flutter/issues/11965#.

**Pull request merged**